sdk/cosmosdb/cosmos/src/queryExecutionContext/pipelinedQueryExecutionContext.ts

```diff
@@ -261,6 +275,17 @@ export class PipelinedQueryExecutionContext implements ExecutionContext {
         ? queryInfo.offset + queryInfo.limit
         : options["vectorSearchBufferSize"] && options["vectorSearchBufferSize"] > 0
           ? options["vectorSearchBufferSize"]
-          : PipelinedQueryExecutionContext.DEFAULT_VECTOR_SEARCH_BUFFER_SIZE;
+          : PipelinedQueryExecutionContext.DEFAULT_MAX_VECTOR_SEARCH_BUFFER_SIZE;
+  }
+
+  private checkTopOrLimit(queryInfo: QueryInfo): void {
+    if (!queryInfo.top && !queryInfo.limit && queryInfo.top !== 0 && queryInfo.limit !== 0) {
```

Comment:
Can be refactored:

hasTop = queryInfo.top || queryInfo.top === 0;
hasLimit = queryInfo.limit || queryInfo.limit === 0;

Reply:
added